textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -59,6 +72,22 @@ public void open() throws DataFlowException {
             TopDocs topDocs = luceneIndexSearcher.search(dataReaderPredicate.getLuceneQuery(), Integer.MAX_VALUE);
             scoreDocs = topDocs.scoreDocs;
             cursor = OPENED;
+
+            this.queryTokens = Utils.tokenizeQuery(dataReaderPredicate.getAnalyzer(),dataReaderPredicate.getQueryString());
+            // sort the query tokens, as the term vector are also sorted.
+            // This makes the seek faster.
+            this.queryTokens.sort(String.CASE_INSENSITIVE_ORDER);
+
+            this.queryTokensInBytesRef = new ArrayList<>();

```

Comment:
The terms in the term vector are stored as ByteRef, hence convert token from String format to ByteRef and then search.